open_discussions_api/channels/client.py

```diff
@@ -0,0 +1,27 @@
+"""Channels API"""
+import json
+
+from open_discussions_api.base import BaseApi
+
+from open_discussions_api.channels.constants import CHANNEL_ATTRIBUTES, VALID_CHANNEL_TYPES
+
+
+class ChannelsApi(BaseApi):
+    """Channels API"""
+
+    def create(self, **channel):
+        """create a new channel"""
+        if not channel:
+            raise AttributeError("No fields provided to create the channel")
+
+        for key in channel:
+            if key not in CHANNEL_ATTRIBUTES:
+                raise AttributeError("Argument '{}' is not a supported field".format(key))
+
+        if channel['channel_type'] not in VALID_CHANNEL_TYPES:
+            raise AttributeError("Channel type '{}' is not a valid option".format(channel['channel_type']))
+
+        return self.session.post(
+            self.get_url("/channels/"),
+            data=json.dumps(dict(channel))
```

Comment:
question: why not specifying that this is a json request using `json=` instead of `data=`?
http://docs.python-requests.org/en/master/user/quickstart/#more-complicated-post-requests

Reply:
not sure, I was following the model that @rhysyngsun was already using for the Users API. I tried out using `json=dict(channel_params)` and it seems to work fine!

Reply:
yes, they should be equivalent, but `json=` takes care of the serialization.
Note: `channel_params` should be already a dictionary so you do not need to cast it

Reply:
TIL!